puz/Square.hpp

```diff
@@ -187,12 +188,12 @@ class PUZ_API Square
     // SetSolution(Square::Black) also calls SetText(Square::Black) for
     // consistency, so if your puzzle is diagramless, you will have to
     // explicitly call SetText("") on black squares to make them empty.
-    bool IsWhite()         const { return ! IsBlack() && ! IsMissing(); }
-    bool IsBlack()         const { return m_text == Black; }
+    bool IsWhite()         const { return !IsBlack() && !IsMissing(); }
+    bool IsBlack()         const { return m_text == Black || IsClue(); }

```

Comment:
Oh, I didn't understand this initially, but does "clue" here refer to the letters? I had assumed that it meant "squares that were part of a clue", rather than "squares that are part of the quote".

I wonder if more broadly we cold use a concept of an "annotation" square, or a "non-editable" square, since it looks like the heavy lifting in this PR is using black squares essentially for layout and decoration (which is clever -- not something that would have immediately come to my mind).